src/library/rekordbox/rekordboxfeature.cpp

```diff
@@ -0,0 +1,960 @@
+// rekordboxfeature.cpp
+// Created 05/24/2019 by Evan Dekker
+
+#include <QMap>
+#include <QMessageBox>
+#include <QSettings>
+#include <QStandardPaths>
+#include <QtDebug>
+
+#include "library/rekordbox/rekordbox_pdb.h"
+#include "library/rekordbox/rekordboxfeature.h"
+
+#include "library/librarytablemodel.h"
+#include "library/missingtablemodel.h"
+#include "library/queryutil.h"
+#include "library/trackcollection.h"
+#include "library/treeitem.h"
+#include "track/keyfactory.h"
+#include "util/file.h"
+#include "util/sandbox.h"
+#include "waveform/waveform.h"
+
+#include "widget/wlibrary.h"
+#include "widget/wlibrarytextbrowser.h"
+
+#define IS_RECORDBOX_DEVICE "::isRecordboxDevice::"
+#define IS_NOT_RECORDBOX_DEVICE "::isNotRecordboxDevice::"
+
+namespace {
+const QString kPDBPath = "PIONEER/rekordbox/export.pdb";
+const QString kPLaylistPathDelimiter = "-->";
+
+void clearTable(QSqlDatabase &database, QString tableName) {
+    QSqlQuery query(database);
+    query.prepare("delete from " + tableName);
+
+    if (!query.exec()) {
+        LOG_FAILED_QUERY(query)
+                << "tableName:" << tableName;
+    } else {
+        query.prepare("delete from sqlite_sequence where name='" + tableName + "'");
+
+        if (!query.exec()) {
+            LOG_FAILED_QUERY(query)
+                    << "tableName:" << tableName;
+        } else {
+            qDebug() << "Rekordbox table entries of '" << tableName << "' have been cleared.";
+        }
+    }
+}
+
+QList<TreeItem *> findRekordboxDevices(RekordboxFeature *rekordboxFeature) {
```

Comment:
Please add a comment that this function will be executed in a separate thread other than the main thread.